components/builder-api/src/server/framework/middleware.rs

```diff
@@ -100,7 +104,7 @@ fn authenticate(req: &HttpRequest<AppState>, token: &str) -> error::Result<Sessi
     // Check for a valid personal access token
     if bldr_core::access_token::is_access_token(token) {
         let mut memcache = req.state().memcache.borrow_mut();
-        match memcache.get_session(token) {
+        match memcache.get_access_token(token) {
```

Comment:
This is excellent.. I think we can make it even simpler.  Just have a single get_session/set_session memcache API.  If the cache hits, great.  If the cache misses, check whether the token is a personal access token, and if so, do the validation according to the None branch below.  If the token is not a personal access token, then return session expired.